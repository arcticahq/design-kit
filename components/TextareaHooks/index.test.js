import React from "react"
import { mount } from "enzyme"
import Textarea from "."

describe(`Textarea`, () => {
  let props
  let mountedComponent
  const wrapper = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Textarea {...props} />)
    }
    return mountedComponent
  }

  const setState = jest.fn()
  const useStateSpy = jest.spyOn(React, "useState")
  useStateSpy.mockImplementation(init => [init, setState])

  beforeEach(() => {
    props = {
      name: `message`,
      title: `Your message`,
      rows: 6,
      required: true,
      showRequired: () => true,
      isPristine: () => true,
      showError: jest.fn(),
      getErrorMessage: jest.fn(),
      setValue: jest.fn(),
      getValue: jest.fn(),
      isValid: jest.fn(),
      isFormSubmitted: jest.fn()
    }
    mountedComponent = undefined
  })

  it(`renders an Textarea correctly`, () => {
    expect(wrapper()).toMatchSnapshot()
  })

  it("Checks for errors when the input is blurred", () => {
    wrapper()
      .find("textarea")
      .simulate("focus")
      .simulate("blur")
    // expect(setState).toHaveBeenCalledWith(true)
    expect(props.getErrorMessage).toBeCalled()
  })

  // shows an error when invalid

  // clears errror when valid

  it("Calls setValue() when the input is changed", () => {
    wrapper()
      .find("textarea")
      .simulate("change", { target: { value: "Some message" } })
    expect(props.setValue).toBeCalled()
  })

  // does not contain required label or text when not required
})
