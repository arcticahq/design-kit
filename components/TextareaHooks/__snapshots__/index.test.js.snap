// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders an Textarea correctly 1`] = `
<Textarea
  getErrorMessage={[MockFunction]}
  getValue={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isFormSubmitted={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  isPristine={[Function]}
  isValid={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  name="message"
  required={true}
  rows={6}
  setValue={[MockFunction]}
  showError={[MockFunction]}
  showRequired={[Function]}
  title="Your message"
>
  <label
    className="block uppercase tracking-wide text-black text-xs font-bold"
    htmlFor="message"
  >
    Your message
    <TextareaAutosize
      aria-invalid={false}
      aria-required={true}
      className="appearance-none h-20 mt-2 block w-full bg-gray-100 text-gray-800 border rounded-lg py-3 px-3 mb-3 leading-tight focus:outline-none focus:bg-white focus:border-yellow-500"
      id="message"
      inputRef={[Function]}
      minRows={6}
      name="message"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onHeightChange={[Function]}
      required={true}
      useCacheForDOMMeasurements={false}
      value=""
    >
      <textarea
        aria-invalid={false}
        aria-required={true}
        className="appearance-none h-20 mt-2 block w-full bg-gray-100 text-gray-800 border rounded-lg py-3 px-3 mb-3 leading-tight focus:outline-none focus:bg-white focus:border-yellow-500"
        id="message"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        style={
          Object {
            "height": 0,
          }
        }
        value=""
      />
    </TextareaAutosize>
  </label>
</Textarea>
`;
